lib/pure/strmisc.nim

```diff
@@ -85,6 +86,110 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+func parseFloatThousandSep*(str: string; sep = ','; decimalDot = '.'): float {.since: (1, 5).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - String must not be empty.
+  ## - String must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - No separator before a digit.
+  ## - First separator can be anywhere after first digit, but no more than 3 characters.
+  ## - There has to be 3 digits between successive separators.
+  ## - There has to be 3 digits between the last separator and the decimal dot.
+  ## - No separator after decimal dot.
+  ## - No duplicate separators.
+  ## - Floats without separator allowed.
+  ##
+  ## See also:
+  ## * `strutils <strutils.html>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("0") == 0.0
+    doAssert parseFloatThousandSep("-0") == -0.0
```

Comment:
put everything that's obvious (for humans) but that still needs testing into tests/stdlib/tstrmisc.nim

this includes things like `doAssert parseFloatThousandSep("-0.0") == -0.0` (note that -0.0 == 0.0)